src/microbe_stage/Microbe.Interior.cs

```diff
@@ -1022,4 +1096,190 @@ private void HandleChemoreceptorLines(float delta)
         // TODO: should this be cleared each time or only when the chemoreception update interval has elapsed?
         activeCompoundDetections.Clear();
     }
+
+    /// <summary>
+    ///   Absorbs compounds/nutrients from ingested objects.
+    /// </summary>
+    private void HandleDigestion(float delta)
+    {
+        if (Dead)
+            return;
+
+        var compounds = SimulationParameters.Instance.GetAllCompounds();
+        var oxytoxy = SimulationParameters.Instance.GetCompound("oxytoxy");
+
+        // Handle logic if the objects that are being digested are the ones we have engulfed
+        for (int i = engulfedObjects.Count - 1; i >= 0; --i)
```

Comment:
Why is there a manual loop here?

Reply:
Removing items while in a loop. I can use `ToList` here if that's not too expensive.